data/src/test/java/org/apache/iceberg/data/TestMetricsRowGroupFilterTypes.java

```diff
@@ -180,7 +180,7 @@ public void createOrcInputFile(List<Record> records) throws IOException {
     OutputFile outFile = Files.localOutput(ORC_FILE);
     try (FileAppender<Record> appender = ORC.write(outFile)
         .schema(FILE_SCHEMA)
-        .createWriterFunc(GenericOrcWriter::buildWriter)
+        .createWriterFunc(typeDesc -> GenericOrcWriter.buildWriter(FILE_SCHEMA, typeDesc))
```

Comment:
Since the schema is passed to the write builder, what about adding a `createWriterFunc` method that accepts `BiFunction<Schema, TypeDescription>`? Then this wouldn't need to change.

We do this in Avro: https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/avro/Avro.java#L199-L203

That would cut down on the number of files that need to change in this PR.

Reply:
OK, that sounds good to me.

Reply:
Would it make more sense to add `createWriterFunc` method that accepts `BiFunction<Schema, TypeDescription>` instead of replacing the existing one? Replacing the existing `createWriterFunc`  causes changes in the files?
```
TestSparkOrcReadMetadataColumns.java
TestSparkOrcReader.java
TestOrcWrite.java
SparkAppenderFactory.java
```